core/src/main/java/org/lflang/util/ImportUtil.java

```diff
@@ -0,0 +1,99 @@
+   * @return The constructed package URI as a string.
+   * @throws IllegalArgumentException if the URI string does not contain both library and file
+   *     names.
+   */
+  public static String buildPackageURI(String uriStr, Resource resource) {
+    URI currentURI = URI.createURI(".");
+    URI rootURI = currentURI.resolve(resource.getURI());
+    StringBuilder pathBuilder = new StringBuilder();
+
+    String[] uriParts = uriStr.split("/");
+
+    if (uriParts.length < 2) {
+      throw new IllegalArgumentException("URI must contain both library name and file name.");
+    }
+
+    // Traverse upwards until we reach the "src/" directory
+    while (!rootURI.toString().endsWith("src/")) {
+      currentURI = URI.createURI("..");
+      rootURI = currentURI.resolve(resource.getURI());
+      pathBuilder.append("../");
+    }
+
+    pathBuilder
+        .append("../target/lfc_include/")
+        .append(uriParts[0])
+        .append("/src/lib/")
+        .append(uriParts[1]);
+
+    return pathBuilder.toString();
+  }
+
+  /**
+   * Builds a package URI based on the provided URI string and source path. This method works
+   * similarly to the `buildPackageURI`, but it accepts a direct source path instead of a resource.
+   * It traverses upwards to locate the "src/" directory and then constructs the URI pointing to the
+   * library file.
+   *
+   * @param uriStr A string representing the URI of the file. It must contain both the library name
+   *     and file name, separated by a '/'.
+   * @param src The source path from which the URI resolution should start.
+   * @return The constructed package URI as a string.
+   * @throws IllegalArgumentException if the URI string or source path is null, empty, or does not
+   *     contain both the library name and file name.
+   */
+  public static String buildPackageURIfromSrc(String uriStr, String src) {
+    if (uriStr == null || src == null || uriStr.trim().isEmpty() || src.trim().isEmpty()) {
+      throw new IllegalArgumentException("URI string and source path must not be null or empty.");
+    }
+
+    String[] uriParts = uriStr.trim().split("/");
+
+    if (uriParts.length < 2) {
+      throw new IllegalArgumentException("URI must contain both library name and file name.");
+    }
+
+    // Use the src path to create a base path
+    Path rootPath = Paths.get(src).toAbsolutePath();
+
+    // Traverse upwards until we reach the "src/" directory
+    while (!rootPath.endsWith("src")) {
+      rootPath = rootPath.getParent();
+      if (rootPath == null) {
+        throw new IllegalArgumentException("The 'src' directory was not found in the given path.");
+      }
+    }
+
+    Path finalPath =
+        rootPath
+            .resolveSibling("target/lfc_include/")
+            .resolve(uriParts[0].trim()) // library name
+            .resolve("src/lib")
+            .resolve(uriParts[1].trim()); // file name
+
+    return finalPath.toString();
```

Comment:
Why convert back to a string? It seems you convert it back to a `Path` again at the call site. Just let the method return `Path`?

Reply:
The method is invoked [here](https://github.com/lf-lang/lingua-franca/blob/0b94151d072c392b44b3c4819ef6d3895ad990ed/core/src/main/java/org/lflang/federated/generator/FedImportEmitter.java#L35C15-L43C41), where the returned string serves as an argument for the `resolve` function. While there is an overloaded `resolve` method that accepts `Path` objects, I opted to return a string because of the following code:

```java
Path importPath =
    fileConfig
        .srcPath
        .resolve(
            i.getImportURI() != null
                ? i.getImportURI()
                : ImportUtil.buildPackageURIfromSrc(
                    i.getImportPackage(), fileConfig.srcPath.toString()))
        .toAbsolutePath();
```

Since `i.getImportURI()` returns a string, I chose to return a string as well. Alternatively, if we want to return a `Path` and use the `resolve` method that takes `Path` objects, I would need to convert `i.getImportURI()` to a `Path` object.

Reply:
Yes, converting a importURI to a Path is trivial: Paths.get(), so that seems like a good solution to me.